server/events/project_command_runner.go

```diff
@@ -114,6 +108,58 @@ type ProjectCommandRunner interface {
 	ProjectApprovePoliciesCommandRunner
 }
 
+// ProjectCommandRunnerWithJobs is a decorator that creates a new PR status check per project.
```

Comment:
This is sweet, we can use this for potentially all the other status related stuff in the default runner.  I think the name might need some work though, maybe like: `ProjectCommandRunnerOutputWrapper` or like `ProjectOutputWrapper` to keep it concise? wdyt?